File: backend/app/utils/money.py
"""Money formatting helpers. All values are integer paise; 1 rupee = 100 paise."""


def fmt_rupees(paise: int) -> str:
    """Return a ₹-prefixed whole-rupee string with comma separators."""
    sign = "-" if paise < 0 else ""
    rupees = abs(round(paise / 100))
    # Simple comma formatting (works for Indian numbers up to crores in this range)
    formatted = f"{rupees:,}"
    return f"{sign}₹{formatted}"

File: frontend/src/utils/money.js
// Money formatting helpers — mirror of backend/utils/money.py.
// All monetary values are integer paise; never use floats for arithmetic.

export function paiseToRupees(paise) {
  const sign = paise < 0 ? '-' : '';
  const rupees = Math.floor(Math.abs(paise) / 100);
  const paiseRem = Math.abs(paise) % 100;
  return `${sign}₹${rupees.toLocaleString('en-IN')}.${String(paiseRem).padStart(2, '0')}`;
}

export function paiseToRupeesRounded(paise) {
  const sign = paise < 0 ? '-' : '';
  const rupees = Math.round(Math.abs(paise) / 100);
  return `${sign}₹${rupees.toLocaleString('en-IN')}`;
}
